tools/solution_drawer.cpp

```diff
@@ -0,0 +1,147 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <set>
+#include <vector>
+#include <random>
+
+#include <SFML/Graphics.hpp>
+#include <SFML/Window.hpp>
+
+sf::RenderWindow& initialize(double size) {
+  sf::VideoMode vm = sf::VideoMode::getDesktopMode();
+  sf::ContextSettings settings;
+  settings.antialiasingLevel = 0;
+  static sf::RenderWindow window(sf::VideoMode(vm.width / size, vm.height / size), "Vehicle Routing Problem", sf::Style::Titlebar | sf::Style::Close | sf::Style::Resize, settings);
+  window.setPosition(sf::Vector2i(vm.width / 2 - vm.width / (size * 2), vm.height / 2 - vm.height / (size * 2)));
+  window.setKeyRepeatEnabled(1);
+  window.setVerticalSyncEnabled(1);
+  return window;
+}
+
+class SolutionDrawer {
+  static constexpr auto ZOOM_FACTOR = 0.9;
+  static constexpr auto RADIUS      = 3.1;
+public:
+  SolutionDrawer(SolutionDrawer&&) = delete;
+  SolutionDrawer(SolutionDrawer const&) = delete;
+  SolutionDrawer(sf::RenderWindow& window): window {window} {}
+
+  auto solution_parse(std::string const& filename) -> void {
+
+    std::fstream fstream {filename, std::ios_base::in};
+    std::string current_line;
+    std::uint32_t route, vehicle, customer;
+    std::getline(fstream, current_line); // Skipping the solution header.
+
+    while (std::getline(fstream, current_line)) {
+      std::replace(std::begin(current_line), std::end(current_line), ';', ' '); // Replace the commas by spaces.
+      std::istringstream iss (current_line);
+      if (iss >> route >> vehicle >> customer) {
+          route_index.emplace_back(route);
+          customers_id.emplace_back(customer);
+      }
+    }
+
+    routes.resize(std::unique(std::begin(route_index), std::end(route_index)) - std::begin(route_index));
+    for (std::size_t index = 0; index < route_index.size(); ++index) routes[route_index[index]].push_back(customers_id[index]);
```

Comment:
doesn't that ruin the order of customers? anyway, I believe that `std::unique` already does what you need: you can get a sorted vector of unique routes with all the duplicates placed after the iterator that is returned. so the following would do the trick, no additional `push_back`s required:

```cpp
auto unique_end = std::unique(std::begin(routes), std::end(routes));
routes.erase(unique_end, std::end(routes));
```

If there's no comparison for `std::vector<size_t>` you can do one, this is no big deal, I assume

Reply:
correct me if i'm wrong, though: unsure why you need this logic anyway

Reply:
@andrgolubev Indeed, there's an error. Thank you.